pkg/agent/agent_linux_test.go

```diff
@@ -14,6 +14,110 @@
 
 package agent
 
+import (
+	"fmt"
+	"net"
+	"strings"
+	"testing"
+
+	mock "github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"antrea.io/antrea/pkg/agent/config"
+	"antrea.io/antrea/pkg/agent/interfacestore"
+	"antrea.io/antrea/pkg/agent/wireguard"
+	"antrea.io/antrea/pkg/ovs/ovsconfig"
+	ovsconfigtest "antrea.io/antrea/pkg/ovs/ovsconfig/testing"
+)
+
 func mockSetInterfaceMTU(returnErr error) func() {
 	return func() {}
 }
+
+func TestPrepareOVSBridgeForK8sNode(t *testing.T) {
+	macAddr, _ := net.ParseMAC("00:00:5e:00:53:01")
+	_, nodeIPNet, _ := net.ParseCIDR("192.168.10.10/24")
+	ipDevice := &net.Interface{
+		Index:        10,
+		MTU:          1500,
+		Name:         "ens160",
+		HardwareAddr: macAddr,
+	}
+	datapathID := "0000" + strings.Replace(macAddr.String(), ":", "", -1)
+	nodeConfig := &config.NodeConfig{
+		UplinkNetConfig: new(config.AdapterNetConfig),
+		NodeIPv4Addr:    nodeIPNet,
+	}
+
+	tests := []struct {
+		name                  string
+		connectUplinkToBridge bool
+		expectedCalls         func(m *ovsconfigtest.MockOVSBridgeClient)
+		expectedErr           string
+	}{
+		{
+			name: "connectUplinkToBridge is false, do nothing",
+		},
+		{
+			name:                  "failed to set datapath_id",
+			connectUplinkToBridge: true,
+			expectedCalls: func(m *ovsconfigtest.MockOVSBridgeClient) {
+				m.EXPECT().SetDatapathID(datapathID).Return(ovsconfig.InvalidArgumentsError("unable to set datapath_id"))
+			},
+			expectedErr: fmt.Sprintf("failed to set datapath_id %s: err=unable to set datapath_id", datapathID),
+		},
+		{
+			name:                  "local port does not exist, allocate it",
+			connectUplinkToBridge: true,
+			expectedCalls: func(m *ovsconfigtest.MockOVSBridgeClient) {
+				m.EXPECT().SetDatapathID(datapathID).Return(nil)
+				m.EXPECT().GetOFPort(ipDevice.Name, false).Return(int32(0), ovsconfig.InvalidArgumentsError("interface not found"))
+				m.EXPECT().AllocateOFPort(config.UplinkOFPort).Return(int32(2), nil).Times(2)
+			},
+		},
+		{
+			name:                  "uplink interface not found",
+			connectUplinkToBridge: true,
+			expectedCalls: func(m *ovsconfigtest.MockOVSBridgeClient) {
+				m.EXPECT().SetDatapathID(datapathID).Return(nil)
+				m.EXPECT().GetOFPort(ipDevice.Name, false).Return(int32(2), nil)
+				m.EXPECT().GetOFPort(ipDevice.Name+"~", false).Return(int32(2), nil)
+			},
+			expectedErr: "cannot find uplink interface",
+		},
+	}
```

Comment:
I think we need a case that finds the uplink, which is the most common code path. The side effect of this function should also be verified.